File: src-tauri/src/handler/journal.rs
// use crate::db::journal::{
//     delete_journal_entry_from_db, delete_journal_template_from_db, get_journal_entries_by_nestling,
//     get_journal_templates_by_nestling, insert_journal_entry_into_db,
//     insert_journal_template_into_db, update_journal_entry_in_db, update_journal_template_in_db,
// };
// use crate::models::journal::{JournalEntry, JournalTemplate, NewJournalEntry, NewJournalTemplate};
// use crate::utils::db::AppDb;
// use crate::utils::errors::AppResult;

// #[tauri::command]
// pub fn insert_journal_entry(
//     db: tauri::State<AppDb>,
//     data: NewJournalEntry,
// ) -> AppResult<JournalEntry> {
//     insert_journal_entry_into_db(&db, data)
// }

// #[tauri::command]
// pub fn get_journal_entries(
//     db: tauri::State<AppDb>,
//     nestling_id: i64,
// ) -> AppResult<Vec<JournalEntry>> {
//     get_journal_entries_by_nestling(&db, nestling_id)
// }

// #[tauri::command]
// pub fn update_journal_entry(
//     db: tauri::State<AppDb>,
//     id: i64,
//     title: String,
//     content: String,
//     entry_date: String,
// ) -> AppResult<()> {
//     update_journal_entry_in_db(&db, id, title, content, entry_date)
// }

// #[tauri::command]
// pub fn delete_journal_entry(db: tauri::State<AppDb>, id: i64) -> AppResult<()> {
//     delete_journal_entry_from_db(&db, id)
// }

// #[tauri::command]
// pub fn insert_journal_template(
//     db: tauri::State<AppDb>,
//     data: NewJournalTemplate,
// ) -> AppResult<JournalTemplate> {
//     insert_journal_template_into_db(&db, data)
// }

// #[tauri::command]
// pub fn get_journal_templates(
//     db: tauri::State<AppDb>,
//     nestling_id: i64,
// ) -> AppResult<Vec<JournalTemplate>> {
//     get_journal_templates_by_nestling(&db, nestling_id)
// }

// #[tauri::command]
// pub fn update_journal_template(
//     db: tauri::State<AppDb>,
//     id: i64,
//     name: String,
//     content: String,
// ) -> AppResult<()> {
//     update_journal_template_in_db(&db, id, name, content)
// }

// #[tauri::command]
// pub fn delete_journal_template(db: tauri::State<AppDb>, id: i64) -> AppResult<()> {
//     delete_journal_template_from_db(&db, id)
// }

File: src/lib/api/journal.ts
import { invoke } from "@tauri-apps/api/core";
import {
  JournalEntry,
  JournalTemplate,
  NewJournalEntry,
  NewJournalTemplate,
} from "../types/journal";

export async function createJournalEntry(data: NewJournalEntry) {
  return await invoke<JournalEntry>("insert_journal_entry", { data });
}

export async function getJournalEntries(nestlingId: number) {
  return await invoke<JournalEntry[]>("get_journal_entries", { nestlingId });
}

export async function updateJournalEntry(
  id: number,
  title: string,
  content: string,
  entryDate: string,
) {
  return await invoke<void>("update_journal_entry", {
    id,
    title,
    content,
    entryDate,
  });
}

export async function deleteJournalEntry(id: number) {
  return await invoke<void>("delete_journal_entry", { id });
}

export async function createJournalTemplate(data: NewJournalTemplate) {
  return await invoke<JournalTemplate>("insert_journal_template", { data });
}

export async function getJournalTemplates(nestlingId: number) {
  return await invoke<JournalTemplate[]>("get_journal_templates", {
    nestlingId,
  });
}

export async function updateJournalTemplate(
  id: number,
  name: string,
  content: string,
) {
  return await invoke<void>("update_journal_template", { id, name, content });
}

export async function deleteJournalTemplate(id: number) {
  return await invoke<void>("delete_journal_template", { id });
}
